crates/swc_ecma_minifier/src/compress/optimize/inline.rs

```diff
@@ -368,9 +368,10 @@ where
                     ident
                 );
                 self.changed = true;
+
                 self.vars
                     .vars_for_inlining
-                    .insert(ident.to_id(), init.take().into());
+                    .insert(ident.take().to_id(), init.take().into());
```

Comment:
If I'm following correctly, the core change is here. We've identified that the variable is useless, by `take`ing the identifier, later code will recognize it as a dummy?

Awesome.